### PR TITLE
Add loadbalance plugin

### PR DESCRIPTION
**Pull request by jgsqware**
When DNS query returns multiple IP for the domain address, the application/CoreDNS takes the 1st one. the list of received IPs are often in the same order, so the application uses the same IP.

In order the use more uniformly the DNS request again the ips, I proposed to use the `loadbalance round_robin` plugin to do it by default. 

When only one IP is return (most of the a, nothing change. 

rel: https://github.com/giantswarm/adidas/issues/398

### CODE REVIEW

File: helm/kubernetes-coredns-chart/templates/configmap.yaml [hunk 1]
```diff
@@ -22,6 +22,7 @@ data:
           class denial
           class error
         }
+        loadbalance {{ .Values.loadbalance.policy }}
```
Comment (MarcelMue) [inline helm/kubernetes-coredns-chart/templates/configmap.yaml]: Missing `:`

**Review by JosephSalisbury — APPROVED** [review]
Config looks good. Check with releng about the versions and all that jazz.

**Review by MarcelMue — CHANGES_REQUESTED** [review]
Please sync with someone from the team responsible for coredns charts how stable this upgrade is.

File: helm/kubernetes-coredns-chart/templates/configmap.yaml (hunk 1 above)
Comment (jgsqware) [reply helm/kubernetes-coredns-chart/templates/configmap.yaml]: where? 

the final config should look like this

```
  Corefile: |
    .:53 {
        errors
        health
        kubernetes cluster.local 172.18.192.0/22 172.18.128.0/18 {
          pods insecure
          upstream
          fallthrough in-addr.arpa ip6.arpa
        }
        prometheus :9153
        proxy . /etc/resolv.conf
        cache 30
        loadbalance round_robin
    }```

Comment (JosephSalisbury) [reply helm/kubernetes-coredns-chart/templates/configmap.yaml]: Yeah, the other colon is just there to prepend the port

Comment (MarcelMue) [reply helm/kubernetes-coredns-chart/templates/configmap.yaml]: I misread, all good.

**Review by MarcelMue — APPROVED** [review]
LGTM if theo / Tobi approves

File: helm/kubernetes-coredns-chart/Chart.yaml [hunk 2]
```diff
@@ -2,4 +2,4 @@ apiVersion: v1
 appVersion: 1.1.1
 description: A Helm chart for CoreDNS
 name: kubernetes-coredns-chart
-version: 0.2.2-[[ .SHA ]]
+version: 0.2.3-[[ .SHA ]]
```
Comment (TheoBrigitte) [inline helm/kubernetes-coredns-chart/Chart.yaml]: There is no need to bump the patch version here.

**Review by TheoBrigitte — APPROVED** [review]

File: helm/kubernetes-coredns-chart/values.yaml [hunk 3]
```diff
@@ -38,6 +38,10 @@ cluster:
       IP: 172.31.0.0
     domain: cluster.local
 
+loadbalance:
+  policy: round_robin
+
+
```
Comment (indyfree) [inline helm/kubernetes-coredns-chart/values.yaml]:
```suggestion
```

**Review by indyfree — APPROVED** [review]

File: helm/kubernetes-coredns-chart/Chart.yaml (hunk 2 above)
Comment (indyfree) [reply helm/kubernetes-coredns-chart/Chart.yaml]: Correct, but also doesn't hurt :+1:

File: helm/kubernetes-coredns-chart/values.yaml (hunk 3 above)
Comment (indyfree) [reply helm/kubernetes-coredns-chart/values.yaml]: Also helm suggest flat values, especially if there is only one. I.e. using `loadbalancePolicy` here. But not critical: 
https://docs.helm.sh/chart_best_practices/#flat-or-nested-values

**Pull request merged**